gap/PGMatrix.gi

```diff
@@ -262,102 +215,229 @@ function(symmetries, fulltgFp, cell, quotient)
 	 Add(PGMLst, [PGMatrixRaw[1], matInt]);
         od;
     fi;
- return [signature, quotient, PGMLst];
+
+    F := NewFamily( "PGMatrixElements", IsPGMatrixElementsObj );
+    return Objectify( NewType( F, IsPGMatrixElementsObj and IsPGMatrixElementsComponentRep ), rec(
+	    signature := MakeImmutable(signature),
+	    quotient := MakeImmutable(TGQuotientName(tgquotient)),
+	    sparse := MakeImmutable(sparse),
+	    elements := MakeImmutable(PGMLst)
+	    ));
 end );
 
 
-InstallGlobalFunction( ExportPGMatrixStream,
-function(pgMatrix, output)
-	local pgMat, item;
-	
-	# Check input:
-	# ------------
 
-	if not IsPGMatrix@(pgMatrix) then
-		Error("the format of the first argument does not match a PGMatrix.");
-		return fail;
-	fi;
+# Type PGMatrix
+DeclareRepresentation("IsPGMatrixComponentRep", IsComponentObjectRep,
+	[ "PGMatrixElements", "PGMatrices" ]
+);
 
-	if not IsOutputTextStream(output) then
-		Error("the provided stream is not a valid output text stream.");
-		return fail;
-	fi;
+InstallMethod( \=, [
+    IsPGMatrixObj and IsPGMatrixComponentRep,
+    IsPGMatrixObj and IsPGMatrixComponentRep ], 
+function(pgMatrix1, pgMatrix2)
+	return 	GetPGMatrixElements(pgMatrix1) = GetPGMatrixElements(pgMatrix2) and
+		Matrices(pgMatrix1) = Matrices(pgMatrix2);
+end );
 
-	# -----------------
-	# Proceed as usual:
-	# -----------------
 
-	SetPrintFormattingStatus(output, false);
+InstallMethod( GetPGMatrixElements, [ IsPGMatrixObj and IsPGMatrixComponentRep ], 
+function(pgMat)
+	return pgMat!.pgMatElements;
+end );
+
+InstallMethod( Matrices, [ IsPGMatrixObj and IsPGMatrixComponentRep ], 
+function(pgMat)
+	return pgMat!.pgMatrices;
+end );
 
-	if NestingDepthA(pgMatrix) = 2 then
-		for item in pgMatrix do
-			AppendTo(output, item);
-			AppendTo(output, "  \n");
-		od;
-	else
-	   	for pgMat in pgMatrix do
-			for item in pgMat do
-				AppendTo(output, item);
-				AppendTo(output, "  \n");
-			 od;
-			AppendTo(output, "\n\n");
-		od;
-	fi;
 
+InstallMethod( PrintString, [ IsPGMatrixObj and IsPGMatrixComponentRep ], 
+function(pgMat)
+	return Concatenation( "PGMatrix( ", 
+		PrintString(GetPGMatrixElements(pgMat)), ", ",
+		"matrices = ", PrintString(Matrices(pgMat)), 
+	")" );
 end );
 
 
-InstallGlobalFunction( ExportPGMatrix,
-function(pgMatrix, path)
-	local output;
 
-	# Check input:
-	# ------------
+InstallGlobalFunction( PGMatrix,
+function(symmetries, fulltg, tg, pgMatElements)	
```

Comment:
According to my proposal above, fulltg and tg would already be stored in pgMatElements which would probably be called something like pgMatRep or similar.